tests/unit/amo/components/TestAddonQRCodeLink.js

```diff
@@ -0,0 +1,164 @@
+    });
+
+    it('does not display a link for a QR code on desktop, for an invalid add-on', () => {
+      const { store } = dispatchClientMetadata({
+        clientApp: CLIENT_APP_FIREFOX,
+      });
+      const root = render({
+        _config,
+        addon: createInternalAddonWithLang({
+          ...fakeAddon,
+          id: goodAddonId + 1,
+        }),
+        store,
+      });
+
+      expect(root.find('.AddonQRCodeLink')).toHaveLength(0);
+    });
+
+    it('does not display a link for a QR code on mobile, for an applicable add-on', () => {
+      const { store } = dispatchClientMetadata({
+        clientApp: CLIENT_APP_ANDROID,
+      });
+      const root = render({ _config, addon: goodAddon, store });
+
+      expect(root.find('.AddonQRCodeLink')).toHaveLength(0);
+    });
+
+    it('displays a modal when user clicks the QRCode link', () => {
+      const preventDefaultSpy = sinon.spy();
+      const { store } = dispatchClientMetadata({
+        clientApp: CLIENT_APP_FIREFOX,
+      });
+      const root = render({ _config, addon: goodAddon, store });
+
+      expect(root.find('.AddonQRCodeLink-modal')).toHaveLength(0);
+
+      root
+        .find('.AddonQRCodeLink-button')
+        .simulate(
+          'click',
+          createFakeEvent({ preventDefault: preventDefaultSpy }),
+        );
+
+      sinon.assert.called(preventDefaultSpy);
+
+      const modal = root.find('.AddonQRCodeLink-modal');
+
+      expect(modal).toHaveLength(1);
+      expect(modal).toHaveProp('id', 'AddonQRCodeLink-modal');
+
+      expect(root.find(AddonQRCode)).toHaveLength(1);
+      expect(root.find(AddonQRCode)).toHaveProp('addon', goodAddon);
+      expect(root.find(AddonQRCode)).toHaveProp(
+        'onDismiss',
+        root.instance().onHideQRCode,
+      );
+    });
+
+    it('closes the modal when user clicks the dismiss button', () => {
+      const preventDefaultSpy = sinon.spy();
+      const { store } = dispatchClientMetadata({
+        clientApp: CLIENT_APP_FIREFOX,
+      });
+      const root = render({ _config, addon: goodAddon, store });
+
+      expect(root.find('.AddonQRCodeLink-modal')).toHaveLength(0);
+
+      root
+        .find('.AddonQRCodeLink-button')
+        .simulate(
+          'click',
+          createFakeEvent({ preventDefault: preventDefaultSpy }),
+        );
+
+      sinon.assert.called(preventDefaultSpy);
+
+      expect(root.find('.AddonQRCodeLink-modal')).toHaveLength(1);
+
+      // Simulate clicking on the dismiss button.
+      root.instance().onHideQRCode();
+
+      expect(root.find('.AddonQRCodeLink-modal')).toHaveLength(0);
+    });
+  });
+
+  describe('flag disabled', () => {
+    const _config = getFakeConfig({
+      ...testConfig,
+      enableFeatureAddonQRCode: false,
+    });
+
+    it('does not display a link for a QR code on desktop, for an applicable add-on', () => {
+      const { store } = dispatchClientMetadata({
+        clientApp: CLIENT_APP_FIREFOX,
```

Comment:
Yes, I thought of that, but I know we generally like to be explicit in tests. I guess adding a comment to `render` that points out this is the default would help. I actually think that having a store with default values lends itself more to the global store approach, because that makes it even more obvious that we are creating a store with default values, as opposed to having the default buried in the `render` method, but I don't want to dredge up that argument again!